src/main.cpp

```diff
@@ -2540,16 +2631,35 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
+            if (!viewTempCache.HaveInputs(tx)) {
+                // ** In the case of IBD we have to check if the height has changed while we're checking inputs because this thread can continue for a
+                // short while after it has been interrupted and meanwhile the view of the UTXO could have changed after another thread has updated the tip.
+                // ** Additionally and more importantly in the event of a big block DDOS attack, by checking the chain height here, we ensure that
+                //    the thread that is validating the big block will immediately exit and finish up (while still keeping the block on disk in the
+                //    case of a reorg) as soon as the first block makes it through and wins the validation race.
+                if (fParallel)
+                {
+                    if (chainActive.Height() > nStartingHeight) {
```

Comment:
@dgenr8 Perhaps your right in that I should at least change this for the sake of consistency and understandability to POW...also I think we should be checking here for not just for  (pow > startingPOW) but also for (pow < startingPOW) as would happen if a re-org were underway while validating this block  (even though we will terminate the thread anyway when tip is disconnected)